### PR TITLE
add aws credentials for system tests

### PR DESCRIPTION
**Pull request by wconti27**
## Summary of changes

necessary for new system tests using AWS resources

## Reason for change

tests r good, tests r fun,

### CODE REVIEW

**Review by bouwkast — APPROVED** [review]
Thanks @wconti27!

**Comment by andrewlock** [comment]
## Execution-Time Benchmarks Report :stopwatch:

Execution-time results for samples comparing the following branches/commits:
- [This PR (6040)](https://github.com/DataDog/dd-trace-dotnet/tree/fd47085c957129c03aff6b98377573d4f5201339)
- [master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)

Execution-time benchmarks measure the whole time it takes to execute a program. And are intended to measure the one-off costs. Cases where the execution time results for the PR are worse than latest master results are shown in **red**. The following thresholds were used for comparing the execution times:
* Welch test with statistical test for significance of **5%**
* Only results indicating a difference greater than **5%** and **5 ms** are considered.


Note that these results are based on a _single_ point-in-time result for each branch. For full results, see the [dashboard](https://ddstaging.datadoghq.com/dashboard/4qn-6fi-54p/apm-net-execution-time-benchmarks).

Graphs show the p99 interval based on the mean and StdDev of the test run, as well as the mean value of the run (shown as a diamond below the graph).
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (6040) - mean (70ms)  : 66, 74
     .   : milestone, 70,
    master - mean (69ms)  : 67, 72
     .   : milestone, 69,

    section CallTarget+Inlining+NGEN
    This PR (6040) - mean (1,117ms)  : 1098, 1135
     .   : milestone, 1117,
    master - mean (1,117ms)  : 1090, 1144
     .   : milestone, 1117,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (6040) - mean (108ms)  : 106, 111
     .   : milestone, 108,
    master - mean (109ms)  : 105, 114
     .   : milestone, 109,

    section CallTarget+Inlining+NGEN
    This PR (6040) - mean (809ms)  : 789, 830
     .   : milestone, 809,
    master - mean (812ms)  : 784, 840
     .   : milestone, 812,

```
```mermaid
gantt
    title Execution time (ms) FakeDbCommand (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (6040) - mean (93ms)  : 88, 98
     .   : milestone, 93,
    master - mean (93ms)  : 88, 99
     .   : milestone, 93,

    section CallTarget+Inlining+NGEN
    This PR (6040) - mean (764ms)  : 743, 785
     .   : milestone, 764,
    master - mean (764ms)  : 739, 788
     .   : milestone, 764,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Framework 4.6.2) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (6040) - mean (191ms)  : 188, 194
     .   : milestone, 191,
    master - mean (190ms)  : 187, 194
     .   : milestone, 190,

    section CallTarget+Inlining+NGEN
    This PR (6040) - mean (1,191ms)  : 1164, 1219
     .   : milestone, 1191,
    master - mean (1,194ms)  : 1172, 1217
     .   : milestone, 1194,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET Core 3.1) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (6040) - mean (276ms)  : 270, 283
     .   : milestone, 276,
    master - mean (277ms)  : 272, 283
     .   : milestone, 277,

    section CallTarget+Inlining+NGEN
    This PR (6040) - mean (970ms)  : 948, 992
     .   : milestone, 970,
    master - mean (969ms)  : 949, 989
     .   : milestone, 969,

```
```mermaid
gantt
    title Execution time (ms) HttpMessageHandler (.NET 6) 
    dateFormat  X
    axisFormat %s
    todayMarker off
    section Baseline
    This PR (6040) - mean (265ms)  : 260, 269
     .   : milestone, 265,
    master - mean (265ms)  : 260, 269
     .   : milestone, 265,

    section CallTarget+Inlining+NGEN
    This PR (6040) - mean (948ms)  : 928, 967
     .   : milestone, 948,
    master - mean (946ms)  : 922, 970
     .   : milestone, 946,

```

File: .azure-pipelines/ultimate-pipeline.yml [hunk 1]
```diff
@@ -5053,6 +5053,10 @@ stages:
         env:
           DD_API_KEY: $(SYSTEM_TESTS_DD_API_KEY)
           TEST_LIBRARY: dotnet
+          AWS_ACCESS_KEY_ID: $(SYSTEM_TESTS_IDM_AWS_ACCESS_KEY_ID)
+          AWS_SECRET_ACCESS_KEY: $(SYSTEM_TESTS_IDM_AWS_SECRET_ACCESS_KEY)
+          AWS_REGION: us-east-1
```
Comment (bouwkast) [inline .azure-pipelines/ultimate-pipeline.yml:5058]: I think this one is actually overridden by a current key that was already set, but not set to be overridable - I've updated it to be overridable so it should work 👍

Comment (bouwkast) [reply .azure-pipelines/ultimate-pipeline.yml:5058]: Actually disregard that, I think that is only for being able to override it when running manually

**Review by zacharycmontoya — APPROVED** [review]
LGTM, it looks like accessing the secrets should all be set up 👍🏼

**Comment by andrewlock** [comment]
## Benchmarks Report for tracer :snail:

Benchmarks for #6040 compared to [master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364): 
* All benchmarks have **the same** speed
* 1 benchmarks have **fewer** allocations


The following thresholds were used for comparing the benchmark speeds:
* Mann–Whitney U test with statistical test for significance of **5%**
* Only results indicating a difference greater than **10%** and **0.3 ns** are considered.

Allocation changes below **0.5%** are ignored.


### Benchmark details


<details><summary><strong>Benchmarks.Trace.ActivityBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`StartStopWithChild`|net6.0|7.78μs|44.5ns|324ns|0.0112| 0.00374|0|5.43&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`StartStopWithChild`|netcoreapp3.1|9.85μs|43.4ns|157ns|0.0143| 0.00476|0|5.62&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`StartStopWithChild`|net472|16.1μs|33ns|123ns|1.03| 0.313|0.102|6.06&nbsp;KB|
|#6040|`StartStopWithChild`|net6.0|7.54μs|39.8ns|218ns|0.019| 0.00762|0|5.42&nbsp;KB|
|#6040|`StartStopWithChild`|netcoreapp3.1|9.75μs|53.8ns|309ns|0.0135| 0.00449|0|5.62&nbsp;KB|
|#6040|`StartStopWithChild`|net472|16.3μs|50.9ns|197ns|1.01| 0.296|0.096|6.06&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AgentWriterBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`WriteAndFlushEnrichedTraces`|net6.0|481μs|158ns|549ns|0| 0|0|2.7&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|625μs|304ns|1.18μs|0| 0|0|2.7&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`WriteAndFlushEnrichedTraces`|net472|844μs|342ns|1.33μs|0.419| 0|0|3.3&nbsp;KB|
|#6040|`WriteAndFlushEnrichedTraces`|net6.0|456μs|103ns|370ns|0| 0|0|2.7&nbsp;KB|
|#6040|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|642μs|1.58μs|6.11μs|0| 0|0|2.7&nbsp;KB|
|#6040|`WriteAndFlushEnrichedTraces`|net472|824μs|964ns|3.74μs|0.411| 0|0|3.3&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.AspNetCoreBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`SendRequest`|net6.0|203μs|1.17μs|9.62μs|0.197| 0|0|18.45&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`SendRequest`|netcoreapp3.1|225μs|1.29μs|9.79μs|0.216| 0|0|20.61&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`SendRequest`|net472|0.00235ns|0.000752ns|0.00291ns|0| 0|0|0&nbsp;b|
|#6040|`SendRequest`|net6.0|186μs|1.02μs|5.87μs|0.188| 0|0|18.45&nbsp;KB|
|#6040|`SendRequest`|netcoreapp3.1|219μs|1.17μs|9.08μs|0.215| 0|0|20.61&nbsp;KB|
|#6040|`SendRequest`|net472|0.000251ns|8.88E&#8209;05ns|0.00032ns|0| 0|0|0&nbsp;b|
</details>


<details><summary><strong>Benchmarks.Trace.CIVisibilityProtocolWriterBenchmark</strong> - Same speed :heavy_check_mark: Fewer allocations :tada:



#### Fewer allocations :tada: in #6040

|  Benchmark         | Base Allocated | Diff Allocated | Change   | Change % |
|:----------|-----------:|-----------:|--------:|--------:|
| Benchmarks.Trace.CIVisibilityProtocolWriterBenchmark.WriteAndFlushEnrichedTraces&#8209;netcoreapp3.1 | 41.87 KB | 41.61 KB | -262 B | -0.63%

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`WriteAndFlushEnrichedTraces`|net6.0|562μs|2.9μs|16.2μs|0.558| 0|0|41.65&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|697μs|3.34μs|13.8μs|0.326| 0|0|41.87&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`WriteAndFlushEnrichedTraces`|net472|862μs|2.83μs|11μs|8.45| 2.53|0.422|53.29&nbsp;KB|
|#6040|`WriteAndFlushEnrichedTraces`|net6.0|583μs|2.76μs|11μs|0.576| 0|0|41.77&nbsp;KB|
|#6040|`WriteAndFlushEnrichedTraces`|netcoreapp3.1|677μs|3.05μs|12.6μs|0.351| 0|0|41.61&nbsp;KB|
|#6040|`WriteAndFlushEnrichedTraces`|net472|879μs|4.09μs|15.8μs|8.36| 2.64|0.44|53.29&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.DbCommandBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`ExecuteNonQuery`|net6.0|1.27μs|0.95ns|3.68ns|0.014| 0|0|1.02&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`ExecuteNonQuery`|netcoreapp3.1|1.73μs|1.25ns|4.84ns|0.0131| 0|0|1.02&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`ExecuteNonQuery`|net472|2.04μs|2.14ns|8.31ns|0.156| 0|0|987&nbsp;B|
|#6040|`ExecuteNonQuery`|net6.0|1.22μs|0.709ns|2.74ns|0.0146| 0|0|1.02&nbsp;KB|
|#6040|`ExecuteNonQuery`|netcoreapp3.1|1.8μs|1.41ns|5.47ns|0.0131| 0|0|1.02&nbsp;KB|
|#6040|`ExecuteNonQuery`|net472|2.14μs|2.61ns|9.78ns|0.157| 0|0|987&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.ElasticsearchBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`CallElasticsearch`|net6.0|1.23μs|0.454ns|1.7ns|0.0136| 0|0|976&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`CallElasticsearch`|netcoreapp3.1|1.56μs|1.79ns|6.7ns|0.0132| 0|0|976&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`CallElasticsearch`|net472|2.61μs|2.1ns|8.13ns|0.158| 0.0013|0|995&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`CallElasticsearchAsync`|net6.0|1.35μs|0.955ns|3.58ns|0.0135| 0|0|952&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`CallElasticsearchAsync`|netcoreapp3.1|1.65μs|0.667ns|2.41ns|0.0133| 0|0|1.02&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`CallElasticsearchAsync`|net472|2.64μs|1.14ns|4.11ns|0.166| 0.00133|0|1.05&nbsp;KB|
|#6040|`CallElasticsearch`|net6.0|1.12μs|0.818ns|3.17ns|0.0134| 0|0|976&nbsp;B|
|#6040|`CallElasticsearch`|netcoreapp3.1|1.51μs|1.72ns|6.67ns|0.013| 0|0|976&nbsp;B|
|#6040|`CallElasticsearch`|net472|2.41μs|1.01ns|3.76ns|0.158| 0|0|995&nbsp;B|
|#6040|`CallElasticsearchAsync`|net6.0|1.37μs|0.878ns|3.29ns|0.0134| 0|0|952&nbsp;B|
|#6040|`CallElasticsearchAsync`|netcoreapp3.1|1.61μs|0.528ns|1.9ns|0.0137| 0|0|1.02&nbsp;KB|
|#6040|`CallElasticsearchAsync`|net472|2.63μs|2.02ns|7.81ns|0.167| 0|0|1.05&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.GraphQLBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`ExecuteAsync`|net6.0|1.27μs|1.22ns|4.71ns|0.0133| 0|0|952&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`ExecuteAsync`|netcoreapp3.1|1.62μs|0.891ns|3.21ns|0.0129| 0|0|952&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`ExecuteAsync`|net472|1.82μs|2.47ns|9.55ns|0.145| 0|0|915&nbsp;B|
|#6040|`ExecuteAsync`|net6.0|1.23μs|6.38ns|31.9ns|0.0132| 0|0|952&nbsp;B|
|#6040|`ExecuteAsync`|netcoreapp3.1|1.55μs|1.79ns|6.92ns|0.0123| 0|0|952&nbsp;B|
|#6040|`ExecuteAsync`|net472|1.8μs|1.17ns|4.52ns|0.145| 0|0|915&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.HttpClientBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`SendAsync`|net6.0|4.15μs|0.911ns|3.29ns|0.0312| 0|0|2.22&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`SendAsync`|netcoreapp3.1|5.2μs|20.1ns|78ns|0.036| 0|0|2.76&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`SendAsync`|net472|7.84μs|1.8ns|6.98ns|0.496| 0|0|3.15&nbsp;KB|
|#6040|`SendAsync`|net6.0|4.15μs|1.82ns|7.04ns|0.031| 0|0|2.22&nbsp;KB|
|#6040|`SendAsync`|netcoreapp3.1|5.12μs|3.8ns|14.7ns|0.0357| 0|0|2.76&nbsp;KB|
|#6040|`SendAsync`|net472|7.75μs|4.84ns|18.8ns|0.498| 0|0|3.15&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.ILoggerBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`EnrichedLog`|net6.0|1.43μs|0.752ns|2.81ns|0.0229| 0|0|1.64&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`EnrichedLog`|netcoreapp3.1|2.25μs|1.17ns|4.36ns|0.0225| 0|0|1.64&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`EnrichedLog`|net472|2.53μs|0.603ns|2.26ns|0.25| 0|0|1.57&nbsp;KB|
|#6040|`EnrichedLog`|net6.0|1.42μs|0.485ns|1.88ns|0.0228| 0|0|1.64&nbsp;KB|
|#6040|`EnrichedLog`|netcoreapp3.1|2.13μs|1.08ns|4.19ns|0.0216| 0|0|1.64&nbsp;KB|
|#6040|`EnrichedLog`|net472|2.51μs|0.689ns|2.67ns|0.249| 0|0|1.57&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.Log4netBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`EnrichedLog`|net6.0|113μs|165ns|594ns|0.0572| 0|0|4.28&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`EnrichedLog`|netcoreapp3.1|117μs|98.7ns|382ns|0.0589| 0|0|4.28&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`EnrichedLog`|net472|145μs|54.4ns|211ns|0.654| 0.218|0|4.46&nbsp;KB|
|#6040|`EnrichedLog`|net6.0|112μs|77.2ns|299ns|0.0558| 0|0|4.28&nbsp;KB|
|#6040|`EnrichedLog`|netcoreapp3.1|119μs|152ns|588ns|0| 0|0|4.28&nbsp;KB|
|#6040|`EnrichedLog`|net472|146μs|38.4ns|144ns|0.656| 0.219|0|4.46&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.NLogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`EnrichedLog`|net6.0|3.04μs|1.35ns|5.05ns|0.0305| 0|0|2.2&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`EnrichedLog`|netcoreapp3.1|4.31μs|1.12ns|4.36ns|0.0302| 0|0|2.2&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`EnrichedLog`|net472|4.89μs|0.944ns|3.66ns|0.32| 0|0|2.02&nbsp;KB|
|#6040|`EnrichedLog`|net6.0|3.08μs|2.17ns|8.41ns|0.0309| 0|0|2.2&nbsp;KB|
|#6040|`EnrichedLog`|netcoreapp3.1|4.32μs|1.94ns|7.26ns|0.0281| 0|0|2.2&nbsp;KB|
|#6040|`EnrichedLog`|net472|5.02μs|2.02ns|7.83ns|0.319| 0|0|2.02&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.RedisBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`SendReceive`|net6.0|1.38μs|0.582ns|2.18ns|0.0159| 0|0|1.14&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`SendReceive`|netcoreapp3.1|1.77μs|0.981ns|3.8ns|0.015| 0|0|1.14&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`SendReceive`|net472|2.07μs|1.26ns|4.86ns|0.183| 0.00103|0|1.16&nbsp;KB|
|#6040|`SendReceive`|net6.0|1.42μs|0.874ns|3.39ns|0.0162| 0|0|1.14&nbsp;KB|
|#6040|`SendReceive`|netcoreapp3.1|1.68μs|0.866ns|3.24ns|0.0152| 0|0|1.14&nbsp;KB|
|#6040|`SendReceive`|net472|2.12μs|0.714ns|2.77ns|0.183| 0|0|1.16&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SerilogBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`EnrichedLog`|net6.0|2.69μs|10.2ns|39.3ns|0.0225| 0|0|1.6&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`EnrichedLog`|netcoreapp3.1|4.03μs|1.14ns|4.41ns|0.0221| 0|0|1.65&nbsp;KB|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`EnrichedLog`|net472|4.38μs|3.19ns|12.3ns|0.322| 0|0|2.04&nbsp;KB|
|#6040|`EnrichedLog`|net6.0|2.82μs|0.943ns|3.53ns|0.0225| 0|0|1.6&nbsp;KB|
|#6040|`EnrichedLog`|netcoreapp3.1|4.09μs|1.84ns|7.14ns|0.0222| 0|0|1.65&nbsp;KB|
|#6040|`EnrichedLog`|net472|4.41μs|2.06ns|7.72ns|0.324| 0|0|2.04&nbsp;KB|
</details>


<details><summary><strong>Benchmarks.Trace.SpanBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`StartFinishSpan`|net6.0|412ns|0.227ns|0.819ns|0.00803| 0|0|576&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`StartFinishSpan`|netcoreapp3.1|579ns|0.265ns|0.99ns|0.00795| 0|0|576&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`StartFinishSpan`|net472|679ns|0.578ns|2.24ns|0.0918| 0|0|578&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`StartFinishScope`|net6.0|508ns|0.277ns|1.07ns|0.0097| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`StartFinishScope`|netcoreapp3.1|704ns|0.446ns|1.67ns|0.00933| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`StartFinishScope`|net472|856ns|1.78ns|6.89ns|0.104| 0|0|658&nbsp;B|
|#6040|`StartFinishSpan`|net6.0|453ns|0.331ns|1.28ns|0.00815| 0|0|576&nbsp;B|
|#6040|`StartFinishSpan`|netcoreapp3.1|589ns|1.65ns|6.38ns|0.00772| 0|0|576&nbsp;B|
|#6040|`StartFinishSpan`|net472|707ns|0.494ns|1.91ns|0.0917| 0|0|578&nbsp;B|
|#6040|`StartFinishScope`|net6.0|473ns|0.269ns|1.04ns|0.00977| 0|0|696&nbsp;B|
|#6040|`StartFinishScope`|netcoreapp3.1|758ns|0.752ns|2.91ns|0.00953| 0|0|696&nbsp;B|
|#6040|`StartFinishScope`|net472|808ns|0.677ns|2.62ns|0.104| 0|0|658&nbsp;B|
</details>


<details><summary><strong>Benchmarks.Trace.TraceAnnotationsBenchmark</strong> - Same speed :heavy_check_mark: Same allocations :heavy_check_mark:

</summary>

### Raw results


|  Branch |                      Method |     Toolchain |     Mean | StdError |  StdDev |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|---------|---------------------------- |-------------- |---------:|---------:|--------:|-------:|------:|------:|----------:|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`RunOnMethodBegin`|net6.0|653ns|0.484ns|1.88ns|0.00979| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`RunOnMethodBegin`|netcoreapp3.1|895ns|4.29ns|17.1ns|0.00925| 0|0|696&nbsp;B|
|[master](https://github.com/DataDog/dd-trace-dotnet/tree/965a822457f8bf38d077de54cad55227e4eb2364)|`RunOnMethodBegin`|net472|1.12μs|1.12ns|4.33ns|0.104| 0|0|658&nbsp;B|
|#6040|`RunOnMethodBegin`|net6.0|657ns|0.798ns|3.09ns|0.00964| 0|0|696&nbsp;B|
|#6040|`RunOnMethodBegin`|netcoreapp3.1|870ns|0.405ns|1.57ns|0.00941| 0|0|696&nbsp;B|
|#6040|`RunOnMethodBegin`|net472|1.11μs|0.697ns|2.7ns|0.104| 0|0|658&nbsp;B|
</details>

**Pull request merged**